renku/core/workflow/plan.py

```diff
@@ -83,6 +84,18 @@ def remove_plan(name_or_id: str, force: bool, plan_gateway: IPlanGateway, when:
     if latest_version.deleted:
         raise errors.ParameterError(f"The specified workflow '{name_or_id}' is already deleted.")
 
+    composites_containing_child = get_composite_plans_by_child(plan)
+
+    composite_names = "\n\t".join([c.name for c in composites_containing_child])
+
+    if composites_containing_child and not force:
```

Comment:
This can be simplified by using nested ifs:
```
if composites_containing_child:
    if force:
        communication.warn(f"Removing '{name_or_id}', which is still used in these workflows:\n\t{composite_names}")
    else:
    ....
```